runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/EngineContext.java

```diff
@@ -74,6 +75,8 @@ MessageConsumer supplyReceiver(
 
     EventFormatter supplyEventFormatter();
 
+    Consumer<Throwable> supplyReporter();
+
```

Comment:
```suggestion
    void report(
        Throwable ex);

```
Perhaps this is sufficient, since the reporter would be created once per `EngineWorker` and reused for the same core anyway.